torch/nn/functional.py

```diff
@@ -2523,9 +2523,9 @@ def _output_size(dim):
 
         # make scale_factor a tensor in tracing so constant doesn't get baked in
         if torch._C._get_tracing_state():
-            return [(torch.floor(input.size(i + 2) * torch.tensor(float(scale_factors[i])))) for i in range(dim)]
+            return [(torch.floor(float(input.size(i + 2)) * torch.tensor(float(scale_factors[i])))) for i in range(dim)]
         else:
-            return [int(math.floor(int(input.size(i + 2)) * scale_factors[i])) for i in range(dim)]
```

Comment:
This cast was making the input of the operator Floor set as "integer", which is not supported (Floor only accepts float and double as input types), and fails in onnxruntime.

Reply:
I am a bit confused here, seems the tracer won't go with the else branch, why will it affect onnx's export? do you test the script module? But it's equivalent change, so i am okay with it.